src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -781,7 +780,7 @@ BSONObj fillQueryServicePath(const std::vector<std::string>& servicePath)
         nullAdded = true;
       }
 
-      char path[SERVICE_NAME_MAX_LEN * 2];
+      char path[SERVICE_PATH_MAX_TOTAL * 2];
       slashEscape(servicePath[ix].c_str(), path, sizeof(path));
```

Comment:
All clarified on skype.
This is simply a bug fix of a lapsus.
The constant SERVICE_PATH_MAX_TOTAL was created but not used. Instead SERVICE_NAME_MAX_LEN was used (which is meant for service, not servicePath).

NTC